app/models/endpoint.rb

```diff
@@ -3,6 +3,7 @@ class Endpoint < ApplicationRecord
 
   default_value_for :verify_ssl, OpenSSL::SSL::VERIFY_PEER
   validates :verify_ssl, :inclusion => {:in => [OpenSSL::SSL::VERIFY_NONE, OpenSSL::SSL::VERIFY_PEER]}
+  validates :port, :numericality => {:only_integer => true, :allow_nil => true, :greater_than => 0}
```

Comment:
Right.. well, as I mentioned in the PR review, we will need that validation logic in angular too.. and middleware is the only one that does not validate before saving ..

So if you *are* opposed to doing that in the model, I think we could just scratch this, and say that middleware will be fixed once converted :)